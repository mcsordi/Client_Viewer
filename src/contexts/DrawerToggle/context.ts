import { createContext } from 'react';
import { TNavDrawer } from '../../shared/types/NavDrawer';


type TToggle = {
    show: Boolean
    toggleState: () => void
    drawerOptions: TNavDrawer[]
    handleDrawerOptions: (navDrawer: TNavDrawer[]) => void
}

export const toggle = createContext({} as TToggle)
