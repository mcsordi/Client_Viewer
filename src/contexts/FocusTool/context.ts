import { createContext, Dispatch, SetStateAction } from "react";
type TFocusTool = {
    focus: boolean
    setFocus: Dispatch<SetStateAction<boolean>>
}

export const focusTool = createContext({} as TFocusTool)
