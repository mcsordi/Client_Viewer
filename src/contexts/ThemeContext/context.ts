import { createContext } from "react";

interface IThemeProvider {
    light: boolean,
    changeTheme: () => void
}

export const themeContext = createContext({} as IThemeProvider)

