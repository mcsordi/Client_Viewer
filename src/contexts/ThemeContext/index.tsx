import React, { useCallback, useState } from 'react';
import { themeContext } from './context';
import { Children } from '../../shared/types/Children';

export const ThemeContext: React.FC<Children> = ({ children }) => {
  const [light, setLight] = useState<boolean>(true);
  const changeTheme = useCallback(() => {
    setLight((old) => !old);
  }, []);
  return (
    <themeContext.Provider value={{ light, changeTheme }}>
      {children}
    </themeContext.Provider>
  );
};
