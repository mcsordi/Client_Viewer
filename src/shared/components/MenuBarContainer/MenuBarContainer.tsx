import React, { useContext } from 'react';
import { MenuInput } from '../MenuInput/MenuInput';
import { MenuButton } from '../MenuButton/MenuButton';
import { themeContext } from '../../../contexts/ThemeContext/context';
import { MenuButtonTool } from '../MenuButtonTools/MenuButtonTool';
import { MenuContainerTools } from '../MenuContainerTools/MenuContainerTools';
import { tools } from '../MenuButtonTools/Tools';
import { FocusTool } from '../../../contexts/FocusTool/index';
import { TMenuButton } from '../../types/MenuButton';

export const MenuBarContainer: React.FC<TMenuButton> = ({
  visible,
  text,
  tVisible,
}) => {
  const { light } = useContext(themeContext);
  return (
    <FocusTool>
      <div
        className={`${light ? `bg-white` : `bg-zinc-700`} w-full p-3.5 shadow-md min-h-16 rounded-md flex justify-between`}
      >
        {<MenuInput visible={visible} />}
        {<MenuButton visible={visible} text={text} />}
        <MenuContainerTools visible={tVisible}>
          {tools.map((tool) => {
            return (
              <MenuButtonTool
                key={tool.text}
                text={tool.text}
                icon={tool.icon}
                type={tool.type}
                loading={tool.loading}
              />
            );
          })}
        </MenuContainerTools>
      </div>
    </FocusTool>
  );
};
