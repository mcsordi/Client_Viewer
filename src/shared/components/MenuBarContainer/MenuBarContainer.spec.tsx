import { render, screen } from '@testing-library/react';
import { MenuBarContainer } from './MenuBarContainer';
import { themeContext } from '../../../contexts/ThemeContext/context';
import { MemoryRouter } from 'react-router-dom';

describe('<MenuBarContainer/>', () => {
  it('should render a MenuBarComponent without children', () => {
    const { container } = render(
      <MemoryRouter initialEntries={['/pessoas?busca=cavalo']}>
        <MenuBarContainer />
      </MemoryRouter>,
    );
    expect(container.children).toHaveLength(1);
  });
  it('should render a MenuBarComponent with children', () => {
    render(
      <MemoryRouter initialEntries={['/pessoas?busca=cavalo']}>
        <MenuBarContainer visible text="novo" />
      </MemoryRouter>,
    );
    const textButton = screen.getByText('novo');
    expect(textButton).toBeInTheDocument();
    expect(textButton.parentNode?.firstChild).toBeInTheDocument();
  });
  it('should render a MenuBarComponent with white background', () => {
    const changeTheme = () => {};
    render(
      <MemoryRouter initialEntries={['/pessoas?busca=cavalo']}>
        <themeContext.Provider value={{ light: true, changeTheme }}>
          <MenuBarContainer visible text="novo" />
        </themeContext.Provider>
      </MemoryRouter>,
    );
    const textButton = screen.getByText('novo');
    expect(textButton.parentNode?.parentElement).toHaveClass('bg-white');
  });
  it('should render a MenuBarComponent with dark background', () => {
    const changeTheme = () => {};
    render(
      <MemoryRouter initialEntries={['/pessoas?busca=cavalo']}>
        <themeContext.Provider value={{ light: false, changeTheme }}>
          <MenuBarContainer visible text="novo" />
        </themeContext.Provider>
      </MemoryRouter>,
    );
    const textButton = screen.getByText('novo');
    expect(textButton.parentElement?.parentElement).toHaveClass('bg-zinc-700');
  });
  it('should match a snapshot', () => {
    const changeTheme = () => {};
    render(
      <MemoryRouter initialEntries={['/pessoas?busca=cavalo']}>
        <themeContext.Provider value={{ light: false, changeTheme }}>
          <MenuBarContainer visible text="novo" />
        </themeContext.Provider>
      </MemoryRouter>,
    );
    const textButton = screen.getByText('novo');
    expect(textButton).toBeInTheDocument();
    expect(textButton.parentNode).toMatchSnapshot();
  });
});
