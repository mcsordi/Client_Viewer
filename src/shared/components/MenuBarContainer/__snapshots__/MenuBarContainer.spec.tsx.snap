// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuBarContainer/> should match a snapshot 1`] = `
<button
  class="font-roboto text-lg flex items-center gap-2 border bg-slate-500 text-white px-3.5 rounded-md font-bold hover:bg-amber-400 transition-all uppercase"
>
  <span
    class="hidden sm:flex md:hidden lg:flex"
  >
    novo
  </span>
  <svg
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    stroke-width="0"
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M15 2.013H9V9H2v6h7v6.987h6V15h7V9h-7z"
    />
  </svg>
</button>
`;
